lua/obsidian/command.lua

```diff
@@ -294,6 +294,63 @@ command.complete_args = function(client, arg_lead, cmd_line, cursor_pos)
   return completions
 end
 
+--- Follow link under cursor.
+---
+---@param client obsidian.Client
+command.follow = function(client, _)
+  local scan = require "plenary.scandir"
+
+  local open, close = util.cursor_on_markdown_link()
+  local current_line = vim.api.nvim_get_current_line()
+
+  if open == nil or close == nil then
+    return nil
+  end
+
+  local note_name = current_line:sub(open + 2, close - 1)
+  local path = client.dir
+  
+  if note_name:match("|[^%]]*") then
+    note_name = note_name:sub(1, note_name:find('|'))
+  end
+
+  if not note_name:match("%.md") then
+    note_name = note_name .. ".md"
+  end
+  
+
+  local notes = {}
+
+  if not note_name:match("/") then
+    scan.scan_dir(vim.fs.normalize(tostring(client.dir)), {
+      hidden = false,
+      add_dirs = false,
+      only_dirs = true,
+      respect_gitignore = true,
+      on_insert = function(entry)
+          note_dir = entry .. "/" .. note_name
+          ok, _ = os.rename(note_dir, note_dir)
```

Comment:
What's the rename for?

Reply:
From my searching, lua doesn't have a function to check if a file exists built in and I also couldn't find that operation in the plenary plugin. So, it's basically a hack I found on SO to check if a file exists. `ok` will only be true if the file exists and the rename operation successfully renames the file to itself. 

It's definitely possible I missed something, though, and am happy to change it if there's a clearer way. Otherwise, I'll just add a comment.

Reply:
Oh, you can use the `Path` object from plenary like this:

```lua
note_dir = Path:new(entry .. "/" .. note_name)
if note_dir:is_dir() then
    ...
```

Reply:
Done!